### PR TITLE
restore MSVC compatibility

### PR DESCRIPTION
**Pull request by tkelman**
Speaks for itself I think. I probably need to rebuild the MSVC LLVM binaries that I had previously uploaded, since my hacky replacement for llvm-config doesn't seem to work any more.

### CODE REVIEW

File: src/builtins.c [hunk 1]
```diff
@@ -39,6 +39,8 @@ DLLEXPORT void NORETURN jl_error(const char *str)
     jl_throw(jl_new_struct(jl_errorexception_type, msg));
 }
 
+extern int vasprintf(char **str, const char *fmt, va_list ap);
```
Comment (vtjnash) [inline src/builtins.c]: travis didn't complain, but should this have an `#ifdef MSVC`?

File: src/gc.c [hunk 2]
```diff
@@ -41,7 +43,9 @@ typedef struct {
     };
     // Work around a bug affecting gcc up to (at least) version 4.4.7
     // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=36839
+#ifndef _MSC_VER
```
Comment (vtjnash) [inline src/gc.c]: `#defined(_COMPILER_MICROSOFT_)` ?

File: src/builtins.c (hunk 1 above)
Comment (tkelman) [inline src/builtins.c]: Don't think so, we have our own at `src/support/asprintf.c` that we use elsewhere, just not declared in this file yet.

File: src/gc.c (hunk 2 above)
Comment (tkelman) [inline src/gc.c]: For lack of anyone testing icc on Windows, sure

**Pull request merged**